c-pallets/scheduler-credit/src/lib.rs

```diff
@@ -30,15 +30,18 @@ pub struct SchedulerCounterEntry {
 
 impl SchedulerCounterEntry {
 	pub fn increase_block_size(&mut self, block_size: u64) {
-		self.proceed_block_size += block_size;
+		self.proceed_block_size += block_size; // overflows?
 	}
 
 	pub fn increase_punishment_count(&mut self) {
-		self.punishment_count += 1;
+		self.punishment_count += 1; // overflows?
 	}
 
 	pub fn figure_credit_score(&self, total_block_size: u64) -> CreditScore {
 		if total_block_size != 0 {
+			//Be aware you can not use float number in the Runtime
+			// more info here: https://substrate.stackexchange.com/questions/146/can-you-use-floating-point-numbers-or-math-in-the-runtime/147#147
+			// I think here is ok as it seems it is only used by the tests
 			let a = (FULL_CREDIT_SCORE as f64 *
 				(self.proceed_block_size as f64 / total_block_size as f64)) as u32;
```

Comment:
Thank you for reminding me. Here I use Perbill instead to calculate:
https://github.com/CESSProject/cess/pull/88/files#diff-d981858a8745d7f332607695d60b4906ea3f7dd08e693c264df4e9a691a8e1fcR45